File: backend/src/main/java/flightbook/entity/search/SearchEntry.java
package flightbook.entity.search;

import flightbook.entity.leg.TripLeg;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;


@Data
@AllArgsConstructor
@NoArgsConstructor
public class SearchEntry {
    private String airlineId;
    private String airlineName;
    private double price;
    private int fromFlightNo;
    private String flightClass;

    private List<TripLeg> tripLegs;

	private Date tripDepTime;
	private Date tripArrTime;
}

File: frontend/src/app/models/search-entry.ts
import { Leg } from './leg';
import { TripLeg } from './trip-leg';

export class SearchEntry {
  airlineId: string;
  airlineName: string;
  price: number;
  fromFlightNo: number;
  flightClass: string;

  tripLegs: TripLeg[];

  tripDepTime: Date;
  tripArrTime: Date;
}
